src/app/code/community/Zendesk/Zendesk/controllers/Adminhtml/ZendeskController.php

```diff
@@ -346,18 +403,317 @@ public function clearLogAction()
+    }
+    
+    public function getUserAction()
+    {
+        $request = $this->getRequest();
+        $id = $request->getParam('id');
+        
+        $user = Mage::getModel('customer/customer')->load($id);
+        
+        $this->getResponse()->clearHeaders()->setHeader('Content-type','application/json',true);
+        
+        if($user->getId()) {
+            $this->getResponse()->setBody(json_encode(array('success'=>true, 'usr'=> array(
+                'firstname' =>  $user->getFirstname(),
+                'lastname'  =>  $user->getLastname(),
+                'email'     =>  $user->getEmail()
+            ))));
+        } else {
+            $this->getResponse()->setBody(json_encode(array('success'=>false, 'msg'=>Mage::helper('zendesk')->__('User does not exist'))));
+        }
+    }
+        
+    public function getOrderAction()
+    {
+        $request = $this->getRequest();
+        $id= $request->getParam('id');
+        
+        $order = Mage::getModel('sales/order')->load($id);
+        
+        $this->getResponse()->clearHeaders()->setHeader('Content-type','application/json',true);
+        if($order->getId()) {
+            $this->getResponse()->setBody(json_encode(array('success'=>true, 'order'=> array(
+                'number' =>  $order->getIncrementId(),
+            ))));
+        } else {
+            $this->getResponse()->setBody(json_encode(array('success'=>false, 'msg'=>Mage::helper('zendesk')->__('Order does not exist'))));
+        }
+    }
+    
+    public function syncAction()
+    {
+        $this->getResponse()->clearHeaders()->setHeader('Content-type','application/json',true);
+        Mage::log('Synchronization started', null, 'zendesk.log');
+        try {           
             $user = Mage::getModel('zendesk/api_users')->all();
-            Mage::getSingleton('adminhtml/session')->addSuccess(Mage::helper('zendesk')->__('Connection to Zendesk API successful'));
-        } catch(Exception $e) {
-            Mage::getSingleton('adminhtml/session')->addError(Mage::helper('zendesk')->__('Connection to Zendesk API failed') .
-                '<br />' . $e->getCode() . ': ' . $e->getMessage() .
-                '<br />' . Mage::helper('zendesk')->__('Troubleshooting tips can be found at <a href=%s>%s</a>', 'https://support.zendesk.com/entries/26579987', 'https://support.zendesk.com/entries/26579987'));
+            if (is_null($user))
+                throw new Exception("Connection Failed");
+            
+            $data = array();
+            $data[] = array(
+                'user_field' => array(
+                    'type'          =>  'integer',
+                    'title'         =>  'ID',
+                    'description'   =>  'Magento Customer Id',
+                    'position'      =>  0,
+                    'active'        =>  true,
+                    'key'           =>  'id'
+                )
+            );
+            $data[] = array(
+                'user_field' => array(
```

Comment:
Why sync customer name into it's own field, rather than into the default user field?